src/Core/Extensible.php

```diff
@@ -388,6 +388,9 @@ public static function has_extension($classOrExtension, $requiredExtension = nul
             if (!$strict && is_subclass_of($extension, $requiredExtension)) {
                 return true;
             }
+            if (Injector::inst()->get($extension) instanceof $requiredExtension) {
```

Comment:
```suggestion
            if (!$strict && Injector::inst()->get($extension) instanceof $requiredExtension) {
```

Reply:
That should fix the tests. There was a test specifically against this type of checking when `strict` mode is on...

Reply:
It will fix the test, but I'm not sure it's correct. Maybe it's worthy of discussion.

In my mind, `has_extension()` should check for a named implementation, not necessarily a concretion.

I can see the case where "not strict" would allow subclasses, but if `strict` is on, and the service name matches, that to me seems correct.

Reply:
If $strict is true, you should probably do a get_class() comparison on the result of Injector::inst()->get().

Reply:
The docblock of the strict parameter reads:
> if the extension has to match the required extension and not be a subclass
The current patch breaks this functionality. @tractorcow has a good suggestion, we can't just `return true`. Instead we can `return !$strict || get_class([...]) === Injector::inst()->get([...])`.

Reply:
Agreed on @tractorcow 's strict check.